src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -74,27 +76,17 @@ public static int Run(string commandName, ITemplateEngineHost host, ITelemetryLo
             _ = callbacks ?? throw new ArgumentNullException(nameof(callbacks));
             _ = args ?? throw new ArgumentNullException(nameof(args));
 
-            if (!args.Any(x => string.Equals(x, "--debug:ephemeral-hive")))
-            {
-                EnsureEntryMutex(hivePath, host);
-
-                if (!_entryMutex!.WaitOne())
-                {
-                    return -1;
-                }
-            }
+            Command rootCommand = new RootCommand(LocalizableStrings.CommandDescription);
```

Comment:
Is there a benefit to doing this here instead of encapsulating all of this in `CommandFactory.CreateCommand(...)`?